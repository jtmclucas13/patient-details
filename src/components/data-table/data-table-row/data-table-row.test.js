import React from 'react';
import { shallow } from 'enzyme';
import DataTableRow from './data-table-row';

describe('component DataTableRow', () => {
    test('renders cells', () => {
        const cells = [{ value: '123' }];
        const wrapper = shallowRenderDataTableRow();

        expect(wrapper).toMatchSnapshot();
    });

    test('renders empty row if no cells given', () => {
        const wrapper = shallowRenderDataTableRow();

        expect(wrapper).toMatchSnapshot();
    });
});

/**
* Helper function for rendering DataTableRow in tests without boilerplate required props
* @param {Object} [props={}] props for DataTableRow
* @return {ShallowWrapper} enzyme wrapper for DataTableRow
*/
function shallowRenderDataTableRow(props = {}) {
    const propsToUse = {
        cells: [],
        ...props,
    };

    return shallow(<DataTableRow {...propsToUse} />);
}