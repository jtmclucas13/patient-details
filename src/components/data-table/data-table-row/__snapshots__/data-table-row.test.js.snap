// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component DataTableRow renders cells 1`] = `
<tr
  className="data-table-row"
/>
`;

exports[`component DataTableRow renders empty row if no cells given 1`] = `
<tr
  className="data-table-row"
/>
`;
